go/1.7/plan.sh

```diff
@@ -0,0 +1,119 @@
+pkg_name=go
```

Comment:
We've got some [naming conventions](https://github.com/habitat-sh/core-plans/blob/master/CONTRIBUTING.md#package-name-conventions) to handle this very situation 😄. This package should be named `go17` and placed in a directory at the root called `go17`